### ISSUE TITLE
Apply ABL for multiple classes

### ISSUE THREAD

**Issue by moured** [wangchi95/active-boundary-loss#5]
Thanks, @wangchi95 for the great work. I want to apply ABL for **multiple class** segmentation problems. Is it only the C in line 45 (bellow) to be changed to "total number of classes" in this case?

`            partial(class2one_hot, C=1),
`

**Comment by wangchi95** [comment]
Thanks for your interest in our work. Our loss is natively designed for multiple classes, so you can directly feed the target with shape _[N,H,W]_ and logits with shape _[N,C,H,W]_  to ABL, where the _C_ here is the number of classes. In more detail, we obtain GTB ([function in line #89](https://github.com/wangchi95/active-boundary-loss/blob/5820cf377695e678f4bd9c95d0f46235b1a49a98/abl.py#L89)) and PDB ([function in line  #65](https://github.com/wangchi95/active-boundary-loss/blob/5820cf377695e678f4bd9c95d0f46235b1a49a98/abl.py#L65)) by comparison and measuring multi-class probability differences with KL divergence, respectively​.

The [code around line #45](https://github.com/wangchi95/active-boundary-loss/blob/5820cf377695e678f4bd9c95d0f46235b1a49a98/abl.py#L45) is used to generate a distance map via GTB, and the C here is not the classes number of logits.

**Comment by moured** [comment]
Thank you @wangchi95 for the reply.

**Comment by JohnMBrandt** [comment]
On the other hand, are there any changes needed to apply this loss to a binary output?

**Comment by moonfoam** [comment]
> On the other hand, are there any changes needed to apply this loss to a binary output?

No more changes to loss are required for binary output. But it should be noted that the data should be provided in one-hot format. Assuming the shape of the binary probability output is [Batch, H, W], you can use the concatenation of [1-prob, prob] as the prediction input (the shape should be [Batch,2,H,W]) for loss.

**Comment by wangchi95** [comment]
> On the other hand, are there any changes needed to apply this loss to a binary output?

No more changes to loss are required for binary output. But it should be noted that the data should be provided in one-hot format. Assuming the shape of the binary probability output is [Batch, H, W], you can use the concatenation of [1-prob, prob] as the prediction input (the shape should be [Batch,2,H,W]) for loss.

**Comment by hansen2001** [comment]
> > On the other hand, are there any changes needed to apply this loss to a binary output?
> 
> No more changes to loss are required for binary output. But it should be noted that the data should be provided in one-hot format. Assuming the shape of the binary probability output is [Batch, H, W], you can use the concatenation of [1-prob, prob] as the prediction input (the shape should be [Batch,2,H,W]) for loss.

Is there any problem with binary segmentation where the output is [batch_size, 2, H, W] and the label is [batch_size, 1, H, W]?
